docs/cip/README.md

```diff
@@ -4,518 +4,1145 @@ Title: Ouroboros Leios - Greater transaction throughput
+<a name="figure-2" id="figure-2"></a>
+<p>
+  <img src="images/spo-profitability.svg" alt="SPO profitability under Praos, as a function of transaction volume">
+</p>
+
+_Figure 2: SPO profitability under Praos, as a function of transaction volume_
+
+</div>
+
+The Leios protocol specified in this document represents a balance
+between immediate scalability needs and long-term protocol evolution. The
+approach prioritizes practical deployment and ecosystem compatibility while
+establishing the foundation for subsequent protocol versions with higher
+throughput capacity.
+
+## Specification
+
+Leios extends Ouroboros Praos by adding a second operating mode. The protocol switches between these two modes automatically based on network conditions and whether certificates appear in blocks. This lets the protocol keep all of Praos's security properties while processing more transactions when the network can handle it.
+
+<div align="center">
+  <a name="figure-3" id="figure-3"></a>
+  <p name="dual-mode-figure">
+    <img src="images/dual-mode-simplified.svg" alt="Dual-Mode Operation">
+  </p>
+
+  <em>Figure 3: Ouroboros Leios Dual-Mode Operation</em>
+</div>
+
+The protocol operates in two modes:
+
+<a name="conservative-mode"></a>**Conservative Mode (CM)** operates with the same transaction validation approach as today's Cardano protocol. Every transaction is fully validated before being included in a block, providing the same security guarantees. However, the protocol also handles certificate announcements and other Leios infrastructure in preparation for potential throughput increases.
+
+<a name="high-throughput-mode"></a>**High-Throughput Mode (HTM)** activates when the network successfully certifies additional transaction capacity. This mode processes more transactions while keeping consensus security intact.
+
+The protocol switches between modes automatically based on certificate inclusion. When the network can handle distributing and validating larger transaction volumes, it enters High-Throughput mode. When it can't, it stays in or returns to Conservative mode.
+
+This design ensures the protocol never performs worse than today's Cardano protocol while delivering higher throughput when network conditions allow. The mode switches happen automatically according to protocol rules (detailed in the [Mode Transition Rules](#mode-transition-rules) section) without requiring any manual coordination from stake pool operators.
+
+### HTM Protocol Flow
+
+<div align="center">
+  <a name="figure-4" id="figure-4"></a>
+  <p name="protocol-flow-figure">
+    <img src="images/protocol-flow-overview.svg" alt="Leios Protocol Flow">
+  </p>
+
+  <em>Figure 4: Leios Protocol Flow</em>
+</div>
+
+The protocol operates through a five-step process that introduces new block types and validation mechanisms to achieve enhanced throughput.
+
+#### Step 1: Block Production
+
+When a stake pool wins block leadership, it **simultaneously** may create two
```

Comment:
Should we be careful about words like "may"? A more rigorous specification would define SHALL, SHOULD, MUST, MAY, etc. Not sure we need this here.